SonarScanner.Shim/PropertiesWriter.cs

```diff
@@ -234,81 +254,6 @@ private static bool IsAscii(char c)
             return c <= sbyte.MaxValue;
         }
 
-        private void WriteSonarProjectInfo()
-        {
-            AppendKeyValue(sb, SonarProperties.ProjectKey, this.config.SonarProjectKey);
-            AppendKeyValueIfNotEmpty(sb, SonarProperties.ProjectName, this.config.SonarProjectName);
-            AppendKeyValueIfNotEmpty(sb, SonarProperties.ProjectVersion, this.config.SonarProjectVersion);
-            AppendKeyValue(sb, SonarProperties.WorkingDirectory, Path.Combine(this.config.SonarOutputDir, ".sonar"));
-            AppendKeyValue(sb, SonarProperties.ProjectBaseDir, ComputeProjectBaseDir());
-
-            sb.AppendLine();
-
-            sb.AppendLine("# FIXME: Encoding is hardcoded");
-            AppendKeyValue(sb, SonarProperties.SourceEncoding, "UTF-8");
-            sb.AppendLine();
-        }
-
-        /// <summary>
-        /// Appends the sonar.projectBaseDir value. This is calculated as follows:
-        /// 1. the user supplied value, or if none
-        /// 2. the sources directory if running from TFS Build or XAML Build, or
-        /// 3. the common root path of projects, or if there isn't any
-        /// 4. the .sonarqube/out directory
-        /// </summary>
-        private string ComputeProjectBaseDir()
-        {
-            string projectBaseDir = this.config.GetConfigValue(SonarProperties.ProjectBaseDir, null);
-            if (!String.IsNullOrWhiteSpace(projectBaseDir))
-            {
-                return projectBaseDir;
-            }
-
-            projectBaseDir = config.SourcesDirectory;
-            if (!String.IsNullOrWhiteSpace(projectBaseDir))
-            {
-                return projectBaseDir;
-            }
-
-            projectBaseDir = GetCommonRootOfProjects();
-            if (!String.IsNullOrWhiteSpace(projectBaseDir))
-            {
-                return projectBaseDir;
-            }
-
-            return this.config.SonarOutputDir;
-        }
-
-
-        private string GetCommonRootOfProjects()
```

Comment:
I would replace this method by the following code:
```
var projectDirectoryParts = projects
	.Select(p => p.GetProjectDirectory())
	.Select(p => p.Split(Path.DirectorySeparatorChar))
	.ToList();

if (projectDirectoryParts.Count == 0)
{
	return string.Empty;
}

var commonParts = projectDirectoryParts
	.OrderBy(p => p.Length)
	.First()
	.TakeWhile((element, index) => projectDirectoryParts.All(p => p[index] == element))
	.ToArray();

return string.Join(Path.DirectorySeparatorChar.ToString(), commonParts);
```

That's more readable for C# devs.